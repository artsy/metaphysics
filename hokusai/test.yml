version: "2"
services:
  metaphysics:
    command: env NODE_OPTIONS=--max_old_space_size=4096 yarn ci
    extends:
      file: build.yml
      service: metaphysics
    environment:
      - MEMCACHED_URL=metaphysics-memcached:11211
      - CI_PULL_REQUEST=$CI_PULL_REQUEST
      - CIRCLE_PULL_REQUEST=$CIRCLE_PULL_REQUEST
      - CIRCLE_BUILD_NUM=$CIRCLE_BUILD_NUM
      - ENABLE_CONSIGNMENTS_STITCHING=true
    env_file: ../.env.test
    depends_on:
      - metaphysics-memcached
  metaphysics-memcached:
    image: memcached
    ports:
      - 11211:11211
