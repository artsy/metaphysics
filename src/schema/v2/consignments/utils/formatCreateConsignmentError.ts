import {
  GraphQLInt,
  GraphQLNonNull,
  GraphQLObjectType,
  GraphQLString,
} from "graphql"
import { formatGravityError } from "lib/gravityErrorHandler"
import { ResolverContext } from "types/graphql"

export const ConsignmentInquiryMutationErrorName =
  "ConsignmentInquiryMutationError"

export const ConsignmentInquiryErrorType = new GraphQLObjectType<
  any,
  ResolverContext
>({
  name: ConsignmentInquiryMutationErrorName,
  fields: () => ({
    type: {
      type: GraphQLString,
    },
    message: {
      type: new GraphQLNonNull(GraphQLString),
    },
    error: {
      type: GraphQLString,
    },
    statusCode: {
      type: GraphQLInt,
    },
  }),
})

export const formatCreateConsignmentError = (error) => {
  return formatGravityError(error)
}
