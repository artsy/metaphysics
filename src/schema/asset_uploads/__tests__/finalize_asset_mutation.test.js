import { runAuthenticatedQuery } from "test/utils"
import gql from "test/gql"

describe("CreateGeminiEntryForAsset", () => {
  it("creates a submission and returns its new data payload", () => {
    const mutation = gql`
      mutation {
        createGeminiEntryForAsset(
          input: {
            template_key: "convection-staging"
            source_key: "2OSuAo_FAxJGHgGy0H2f-g/Orta2 2.jpg"
            source_bucket: "artsy-media-uploads"
            metadata: { id: 144, _type: "Consignment" }
          }
        ) {
          asset {
            token
          }
        }
      }
    `

    const rootValue = {
      createNewGeminiEntryAssetLoader: () =>
        Promise.resolve({
          token: "zVHJce-Fey3OIsazH8WDTg",
          image_urls: {},
          clientMutationId: "1231",
        }),
    }

    expect.assertions(1)
    return runAuthenticatedQuery(mutation, rootValue).then(data => {
      expect(data).toMatchSnapshot()
    })
  })
})
