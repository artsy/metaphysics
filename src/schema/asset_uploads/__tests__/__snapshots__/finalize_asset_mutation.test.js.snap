// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateGeminiEntryForAsset creates a submission and returns its new data payload 1`] = `
Object {
  "createGeminiEntryForAsset": Object {
    "asset": Object {
      "token": "zVHJce-Fey3OIsazH8WDTg",
    },
  },
}
`;
